eng/Directory.Build.Data.targets

```diff
@@ -33,12 +35,20 @@
     <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0-alpha-004" PrivateAssets="all" IsImplicitlyDefined="true" />
   </ItemGroup>
 
-  <PropertyGroup Condition="'$(SignAssembly)' != 'false'">
+  <!-- If signing is enabled for an SDK library, delay sign using the shared key -->
+  <PropertyGroup Condition="'$(SignAssembly)' != 'false' and '$(IsTestProject)' != 'true' ">
     <SignAssembly>true</SignAssembly>
-    <DelaySign>true</DelaySign>
+    <DelaySign >true</DelaySign>
     <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)MSSharedLibKey.snk</AssemblyOriginatorKeyFile>
   </PropertyGroup>
 
+  <!-- If signing is enabled for a Test library, sign immediately using the test key -->
+  <PropertyGroup Condition="'$(SignAssembly)' != 'false' and '$(IsTestProject)' == 'true'">
```

Comment:
For this special case I think I would delete the line above where you are using `AllowTestAssemblySigning`  and continue to always set SignAssembly=false by default and then instead use a new variable called `SignTestAssembly` and use that in the condition here.

Reply:
Oooh, I like that.  Much cleaner semantics.   Changes incoming.